### ISSUE TITLE
TCP Subscriber Failed to Reconnect Within Single Process [9742]

### ISSUE THREAD

**Issue by RuiChen1998** [eProsima/Fast-DDS#1487]
I was trying to implement TCP subscribers and publishers in my applications. 

I opened my publisher in one process and my subscriber in another process and in the beginning, they successfully matched each other. But then if I keep my TCP publisher alive, and restart my TCP subscriber within the same old process (delete the participant_sub object and new another one), the subscriber would always fail to match with the publisher. 

However, if I terminate the whole subscriber process and start a same new subscriber process, it would match again. I suspect that maybe the TCP connection or resources were not fully cleaned up by the destructor and was cleaned up by the OS after the process termination.

## Expected Behavior

Within one process, everytime I start a TCP subscriber it should match to the publisher.

## Current Behavior

Within one process, TCP subscriber matches to the publisher for the first time it was started and fails to match again if it was closed and started again.

## Steps to Reproduce

1. Start a TCP publisher in Process A, a TCP subscriber in Process B. (Could just be a modified version of HelloworldTCP example)
2. They match and data can be sent and received.
3. Close the TCP subscriber in Process B and then start a new TCP subscriber in Process B.
4. This time they CANNOT match.
5. Terminate Process B. Start a new TCP subscriber in a new Process C.
6. They match and data can be sent and received.

## System information

- Fast-RTPS version: 2.0.0 
- OS: Windows 10, Ubuntu 20
- Network interfaces: eth0, wlan, local(127.0.0.1)
                      

## Additional context

Restart the subscrbier in the same process as follows:
```
    std::cout << "Starting TCP subscriber." << std::endl;
    int samples = 10;

    HelloWorldSubscriber* mysub = new HelloWorldSubscriber();
    if(mysub->init())
    {
        mysub->run(static_cast<uint32_t>(samples));
    }

    delete mysub;
    
    
    std::cout << "Starting TCP subscriber again." << std::endl;

    mysub = new HelloWorldSubscriber();
    if(mysub->init())
    {
        mysub->run(static_cast<uint32_t>(samples));
    }

    delete mysub;
```

## Additional resources
- Wireshark capture 
There is no extra TCP packet captured between the delete of the subscriber and the termination of the process.

**Comment by RuiChen1998** [comment]
Also, I found that if I start TWO identical TCP subscriber in one single process in the beginning, ONLY ONE will match to the publisher and the other will not match. But the TCP connection status looks very normal as follows:
tcp        0      0 127.0.0.1:5678          0.0.0.0:*                    LISTEN           4083/./DDSHelloWorl
tcp        0      0 127.0.0.1:34776        127.0.0.1:5678          ESTABLISHED 4070/./DDSHelloWorl
tcp        0      0 127.0.0.1:34778        127.0.0.1:5678          ESTABLISHED 4070/./DDSHelloWorl
tcp        0      0 127.0.0.1:5678          127.0.0.1:34778        ESTABLISHED 4083/./DDSHelloWorl
tcp        0      0 127.0.0.1:5678          127.0.0.1:34776        ESTABLISHED 4083/./DDSHelloWorl

**Comment by JLBuenoLopez-eProsima** [comment]
Sorry @RuiChen1998,

Fast DDS 2.0.x has reached [end of life](https://github.com/eProsima/Fast-DDS/blob/master/RELEASE_SUPPORT.md#release-support). Also, this issue might have been solved by #2470 which has been backported to current Fast DDS supported versions. I suggest you upgrade to one of these releases and check if the issue is still happening. I am going to close the issue but feel free to reopen if the issue is not solved in the newer versions of Fast DDS .
